packages/eas-cli/src/commands/update/index.ts

```diff
@@ -446,15 +448,17 @@ export default class UpdatePublish extends EasCommand {
       try {
         const platforms = platformFlag === 'all' ? defaultPublishPlatforms : [platformFlag];
         const assets = await collectAssetsAsync({ inputDir: inputDir!, platforms });
-        const { uniqueUploadedAssetCount } = await uploadAssetsAsync(
+        const uploadResults = await uploadAssetsAsync(
           assets,
+          projectId,
           (totalAssets, missingAssets) => {
             assetSpinner.text = `Uploading assets. Finished (${
               totalAssets - missingAssets
             }/${totalAssets})`;
           }
         );
-        uploadedAssetCount = uniqueUploadedAssetCount;
+        uploadedAssetCount = uploadResults.uniqueUploadedAssetCount;
+        assetLimitPerUpdateGroup = uploadResults.assetLimitPerUpdateGroup;
```

Comment:
As I was writing this I considered splitting out the `getAssetLimitAsync` from `uploadAssetsAsync`:
```javascript
        const [uploadResults, assetLimitPerUpdateGroup] = await Promise.all([
          uploadAssetsAsync(...),
          PublishQuery.getAssetLimitAsync(...)
        ]);
```
But I felt the asset limit should be handled within `uploadAssetsAsync` since it's so closely related.